docs/template_plugin/src/template_pattern_transformation.cpp

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "template_pattern_transformation.hpp"
+
+#include <ngraph/opsets/opset3.hpp>
+
+// ! [graph_rewrite:template_transformation_cpp]
+// template_pattern_transformation.cpp
+void MyPatternBasedTransformation::transform() {
+    // Pattern example
+    auto input = std::make_shared<ngraph::opset3::Parameter>(ngraph::element::i64, ngraph::Shape{1});
+    auto shapeof = std::make_shared<ngraph::opset3::ShapeOf>(input);
+
+    ngraph::graph_rewrite_callback callback = [](ngraph::pattern::Matcher& m) {
+        // Transformation code
+        return false;
```

Comment:
I would like to see more real example of transformation, but just a pattern with empty body. Maybe introduce new operation inside a template plugin and implement transformation with this new operation

Reply:
I have added additional transformation code